problem_builder/answer.py

```diff
@@ -164,7 +164,7 @@ def student_input(self):
 
     def mentoring_view(self, context=None):
         """ Render this XBlock within a mentoring block. """
-        context = context or {}
+        context = context.copy() if context else {}
         context['self'] = self
```

Comment:
I trust you that this fixes the issue, but I'm confused by why having `"self"` as a parameter in `context` doesn't mess up mako's `render` function